easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -141,3 +141,18 @@ def is_yeb_format(filename, rawcontent):
             if line.startswith('name: '):
                 isyeb = True
     return isyeb
+
+def quote_yaml_special_chars(val):
+    """
+    Single-quote values that contain special characters, specifically to be used in YAML context (.yeb files)
+    """
+    if isinstance(val, basestring):
+        if "'" in val:
+            if '"' in val:
+                val = val.replace('"', '\\"')
+            val = '"%s"' % val
+
+        elif YAML_SPECIAL_CHARS.intersection(val):
+            val = "'%s'" % val
```

Comment:
@Caylo shouldn't this be the first option you try?

if the string includes both `'` and any special YAML charachter, you'll still be in trouble since then you'll take the first option which quotes with `"..."` but doesn't escape YAML chars?